### ISSUE TITLE
fs.find() returns a list instead of a dictionary

### ISSUE THREAD

**Issue by ldacey** [dask/adlfs#135]
**What happened**:
I am running into errors with ds.dataset() from pyarrow which expects a dictionary instead of a list from fs.find() from adlfs. 

**What you expected to happen**:
The code seems to show that a dictionary should be returned if detail=True, but I wasn't able to get this to work.

**Minimal Complete Verifiable Example**:
dev is my Azure Blob container and testing is a dataset written with pq.write_to_dataset() with two partition columns being used.

```python
fs = fsspec.filesystem(protocol="abfs", account_name=name, account_key=key)

print(type(fs.find("dev/testing", detail=True)))
<class 'list'>

print(type(fs.find("dev/testing")))
<class 'list'>

fs = adlfs.AzureBlobFileSystem(account_name=name, account_key=key)
print(type(fs.find("dev/testing", detail=True)))
<class 'list'>
```

**Anything else we need to know?**:

**Environment**:
I installed this on a fresh PC using conda forge. Python was not installed previously, so I do not think there should be any conflicts with versions.

- Dask version: 2.23 typically but I do **not** have it installed in this example
- Python version: 3.8.6
- Operating System: Ubuntu 20.04
- Install method (conda, pip, source): conda env file

```
name: pyarrow
channels:
  - conda-forge
  - defaults
dependencies:
  - python=3.8
  - pandas
  - numpy
  - pyarrow
  - adlfs
  - azure-storage-blob
```

Full conda list output:

```
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
abseil-cpp                20200225.2           he1b5a44_2    conda-forge
adal                      1.2.5              pyh9f0ad1d_0    conda-forge
adlfs                     0.5.7              pyhd8ed1ab_0    conda-forge
aiohttp                   3.7.3            py38h25fe258_0    conda-forge
arrow-cpp                 2.0.0           py38ha9d6ab6_6_cpu    conda-forge
async-timeout             3.0.1                   py_1000    conda-forge
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
aws-c-common              0.4.59               h36c2ea0_1    conda-forge
aws-c-event-stream        0.1.6                had2084c_6    conda-forge
aws-checksums             0.1.10               h4e93380_0    conda-forge
aws-sdk-cpp               1.8.70               h57dc084_1    conda-forge
azure-core                1.9.0              pyhd3deb0d_0    conda-forge
azure-datalake-store      0.0.51             pyh9f0ad1d_0    conda-forge
azure-identity            1.4.1              pyh9f0ad1d_0    conda-forge
azure-storage-blob        12.6.0             pyhd3deb0d_0    conda-forge
blinker                   1.4                        py_1    conda-forge
brotli                    1.0.9                he1b5a44_3    conda-forge
brotlipy                  0.7.0           py38h8df0ef7_1001    conda-forge
bzip2                     1.0.8                h516909a_3    conda-forge
c-ares                    1.17.1               h36c2ea0_0    conda-forge
ca-certificates           2020.11.8            ha878542_0    conda-forge
certifi                   2020.11.8        py38h578d9bd_0    conda-forge
cffi                      1.14.3           py38h1bdcb99_1    conda-forge
chardet                   3.0.4           py38h924ce5b_1008    conda-forge
cryptography              3.2.1            py38h7699a38_0    conda-forge
dbus                      1.13.6               hfdff14a_1    conda-forge
expat                     2.2.9                he1b5a44_2    conda-forge
fsspec                    0.8.4                      py_0    conda-forge
gettext                   0.19.8.1          hf34092f_1004    conda-forge
gflags                    2.2.2             he1b5a44_1004    conda-forge
glib                      2.66.2               h58526e2_0    conda-forge
glog                      0.4.0                h49b9bf7_3    conda-forge
grpc-cpp                  1.33.2               h1870a98_1    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
isodate                   0.6.0                      py_1    conda-forge
jeepney                   0.6.0              pyhd8ed1ab_0    conda-forge
keyring                   21.5.0           py38h578d9bd_0    conda-forge
krb5                      1.17.2               h926e7f8_0    conda-forge
ld_impl_linux-64          2.35.1               hed1e6ac_0    conda-forge
libblas                   3.9.0                3_openblas    conda-forge
libcblas                  3.9.0                3_openblas    conda-forge
libcurl                   7.71.1               hcdd3856_8    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               hcdb4288_3    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-ng                 9.3.0               h5dbcf3e_17    conda-forge
libgfortran-ng            9.3.0               he4bcb1c_17    conda-forge
libgfortran5              9.3.0               he4bcb1c_17    conda-forge
libglib                   2.66.2               hbe7bbb4_0    conda-forge
libgomp                   9.3.0               h5dbcf3e_17    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
liblapack                 3.9.0                3_openblas    conda-forge
libnghttp2                1.41.0               h8cfc5f6_2    conda-forge
libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
libprotobuf               3.13.0.1             h8b12597_0    conda-forge
libssh2                   1.9.0                hab1572f_5    conda-forge
libstdcxx-ng              9.3.0               h2ae2ef3_17    conda-forge
libthrift                 0.13.0               h5aa387f_6    conda-forge
libutf8proc               2.5.0                h516909a_2    conda-forge
lz4-c                     1.9.2                he1b5a44_3    conda-forge
msal                      1.6.0              pyhd3deb0d_0    conda-forge
msal_extensions           0.2.2              pyh9f0ad1d_0    conda-forge
msrest                    0.6.19             pyh9f0ad1d_0    conda-forge
msrestazure               0.6.2                      py_0    conda-forge
multidict                 4.7.5            py38h1e0a361_2    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
numpy                     1.19.4           py38hf0fd68c_1    conda-forge
oauthlib                  3.0.1                      py_0    conda-forge
openssl                   1.1.1h               h516909a_0    conda-forge
orc                       1.6.5                hd3605a7_0    conda-forge
pandas                    1.1.4            py38h0ef3d22_0    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
pcre                      8.44                 he1b5a44_0    conda-forge
pip                       20.2.4                     py_0    conda-forge
portalocker               1.7.0            py38h32f6830_1    conda-forge
pyarrow                   2.0.0           py38h5703e5b_6_cpu    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyjwt                     1.7.1                      py_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pysocks                   1.7.1            py38h924ce5b_2    conda-forge
python                    3.8.6           h852b56e_0_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.4             pyhd8ed1ab_0    conda-forge
re2                       2020.11.01           h58526e2_0    conda-forge
readline                  8.0                  he28a2e2_2    conda-forge
requests                  2.25.0             pyhd3deb0d_0    conda-forge
requests-oauthlib         1.3.0              pyh9f0ad1d_0    conda-forge
secretstorage             3.2.0            py38h578d9bd_0    conda-forge
setuptools                49.6.0           py38h924ce5b_2    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
snappy                    1.1.8                he1b5a44_3    conda-forge
sqlite                    3.33.0               h4cf870e_1    conda-forge
tk                        8.6.10               hed695b0_1    conda-forge
typing-extensions         3.7.4.3                       0    conda-forge
typing_extensions         3.7.4.3                    py_0    conda-forge
urllib3                   1.25.11                    py_0    conda-forge
wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
yarl                      1.6.3            py38h25fe258_0    conda-forge
zlib                      1.2.11            h516909a_1010    conda-forge
zstd                      1.4.5                h6597ccf_2    conda-forge
```

**Comment by jorisvandenbossche** [comment]
Ah, I spotted the error:

https://github.com/dask/adlfs/blob/54d5826627a25dc8541e35754bf4e9512543a027/adlfs/spec.py#L788-L793

The `**kwargs` passed to `find` are not passed through to `maybe_sync(self._find, ..)`

So the `_find` properly handles the `detail=True` case, but the public `find` method ignores it this way.

**Comment by ldacey** [comment]
Nice! I added a pull request for this as well since it really is a very quick and easy fix.

```
    def find(self, path, maxdepth=None, withdirs=False, **kwargs):
        return maybe_sync(self._find, self, path, maxdepth, withdirs, **kwargs)
```

This fixes my issue with the latest versions of adlfs and azure-storage-blob v12 since I am able to read a dataset now (I was already able to use write_dataset() before).

```
dataset = ds.dataset(source="dev/testing9", filesystem=fs, partitioning="hive")
table = dataset.to_table()
print(table.num_rows)
23566
```

Do you think I should close https://issues.apache.org/jira/browse/ARROW-10517? I have a remaining issue with my production environment with an older version of adlfs and azure-blob-storage v2 which might also be a quick fix, but the existing issue is quite long and confusing.

**Comment by hayesgb** [comment]
@ldacey -- I did not see your PR, but added the fix with #136, which is now in master.  Please let me know if this does not fix your issue.

**Comment by ldacey** [comment]
Thanks, yes your changes resolved this issue.
